src/backend/distributed/operations/shard_rebalancer.c

```diff
@@ -429,6 +432,60 @@ GetShardCost(uint64 shardId, void *voidContext)
 }
 
 
+/*
+ * citus_partitioned_shard_total_size returns the total size of a partitioned table with
+ * its partitions, using the function pg_partition_tree.
+ */
+Datum
+citus_partitioned_shard_total_size(PG_FUNCTION_ARGS)
```

Comment:
open ended question that I don't have a good answer: does the name strictly require to include `shard`  -- or even `citus`? This function calculates the size of the partitioned table, including children. So maybe `citus_partitioned_table_total_size()` or such?

Reply:
Seems reasonable

Reply:
maybe `worker_partitioned_relation_total_size`,  `worker_partitioned_relation_size` and ``worker_partitioned_table_size``